src/ai/susi/server/api/aaa/ApiKeysService.java

```diff
@@ -63,11 +63,12 @@ public JSONObject getDefaultPermissions(UserRole baseUserRole) {
     @GET
     @ApiOperation(httpMethod = "GET", value = "Endpoint to add, modify and delete diffrent API keys used by SUSI")
     @ApiResponses(value = {
-            @ApiResponse(code = 200, message = "Added new API key xxxxxxxx successfully"),
             @ApiResponse(code = 200, message = "Removed API key successfully"),
+            @ApiResponse(code = 201, message = "Added new API key xxxxxxxx successfully"),
             @ApiResponse(code = 400, message = "Bad Request. No parameter present"),
+            @ApiResponse(code = 401, message = "Base user role not sufficient. Your base user role is 'ANONYMOUS', minimum user role required is 'admin'"),
             @ApiResponse(code = 500, message = "Failed: Unable to add xxxxxxxx!"),
-            @ApiResponse(code = 500, message = "Failed: xxxxxxxx doesn't exists")
+            @ApiResponse(code = 501, message = "Failed: xxxxxxxx doesn't exists")
```

Comment:
Is ```501``` correct code for this, since it says ```xxxxxxx doesn't exits``` and not ```Not Implemented```?

Reply:
I thought 5XX error codes are for server errors for the cases in which the server is aware that it has encountered an error or is otherwise incapable of performing the request.
Can you please suggest some other code in the scenario?